python/src/scipp/plot/plot_2d.py

```diff
@@ -211,22 +214,36 @@ def update_axes(self):
         for dim, button in self.buttons.items():
             if self.slider[dim].disabled:
                 but_val = button.value.lower()
+                # xc = self.slider_x[self.name][dim].values
                 if not self.histograms[self.name][dim]:
                     xc = self.slider_x[self.name][dim].values
                     if self.slider_nx[self.name][dim] < 2:
-                        xmin = xc[0] - 0.5
-                        xmax = xc[0] + 0.5
+                        dx = 0.5 * abs(xc[0])
+                        if dx == 0.0:
+                            dx = 0.5
+                        xmin = xc[0] - dx
+                        xmax = xc[0] + dx
+                        axparams[but_val]["xmin"] = xmin
+                        axparams[but_val]["xmax"] = xmax
                     else:
                         xmin = 1.5 * xc[0] - 0.5 * xc[1]
                         xmax = 1.5 * xc[-1] - 0.5 * xc[-2]
                     self.extent[but_val] = [xmin, xmax]
                 else:
                     self.extent[but_val] = self.slider_x[
-                        self.name][dim].values[[0, -1]]
+                        self.name][dim].values[[0, -1]].astype(np.float)
 
+                axparams[but_val]["lims"] = self.extent[but_val].copy()
+                if getattr(self, "log" + but_val) and (
+                    self.extent[but_val][0] <= 0):
+                    if not self.histograms[self.name][dim]:
+                        new_x = centers_to_edges(xc)
+                    else:
+                        new_x = edges_to_centers(self.slider_x[self.name][dim].values)
+                    axparams[but_val]["lims"][0] = new_x[np.searchsorted(new_x, 0)]
```

Comment:
Is this fixing a bug? It looks unrelated?

Reply:
Yes, unrelated. The axes limits for log images were wrong, they should be showing the same range as the 1D plots. See e.g. https://scipp.readthedocs.io/en/latest/visualization/plotting-overview.html#Logarithmic-scale